databallpy/data_parsers/event_data_parsers/statsbomb_parser.py

```diff
@@ -0,0 +1,507 @@
+        "full_name": [],
+        "formation_place": [],
+        "position": [],
+        "starter": [],
+        "shirt_num": [],
+    }
+
+    for player in players_data:
+
+        if len(player["positions"])>0:
+            if player["positions"][0]["from"] == "00:00":
+                starter = True
+            else:
+                starter = False
+            formation_place = player["positions"][0]["position_id"]
+            position = player["positions"][0]["position"].lower()
+        else:
+            starter = False
+            formation_place = MISSING_INT
+            position = "did not play"
+
+
+        result_dict["id"].append(player["player_id"])
+        result_dict["full_name"].append(player["player_name"])
+        result_dict["formation_place"].append(formation_place)
+        result_dict["position"].append(position)
+        result_dict["starter"].append(starter)
+        result_dict["shirt_num"].append(player["jersey_number"])
+
+    return pd.DataFrame(result_dict)
+
+def _load_event_data(events_loc: str, metadata: Metadata, pitch_dimensions: tuple)-> tuple[pd.DataFrame, dict, Metadata]:
+    """This function retrieves the event data of a specific match. The x
+    and y coordinates provided have been scaled to the dimensions of the pitch, with
+    (0, 0) being the center. Additionally, the coordinates have been standardized so
+    that the home team is represented as playing from left to right for the entire
+    match, and the away team is represented as playing from right to left.
+
+    Args:
+        events_loc (str): location of the events.json file
+        metadata (Metadata): metadata of the match
+        pitch_dimensions (tuple): dimensions of the pitch.
+
+    Returns:
+        Tuple[pd.DataFrame, dict, Metadata]: the event data of the match, the databallpy_events, and the updated metadata
+    """
+    with open(events_loc, "r", encoding="utf-8") as f:
+        events_json = json.load(f)
+
+    formations = {
+        event["team"]["id"]: str(event["tactics"]["formation"])
+        for event in events_json if event["type"]["id"] == 35
+    }
+    metadata.home_formation = formations[metadata.home_team_id]
+    metadata.away_formation = formations[metadata.away_team_id]
+
+    event_data = {
+        "event_id": [],
+        "databallpy_event": [],
+        "period_id": [],
+        "minutes": [],
+        "seconds": [],
+        "player_id": [],
+        "team_id": [],
+        "outcome": [],
+        "start_x": [],
+        "start_y": [],
+        "datetime": [],
+        "statsbomb_event": [],
+        "statsbomb_event_id": [],
+        "player_name": [],
+        "team_name": [],
+    }
+    
+    databallpy_mapping = {
+        "Pass": "pass",
+        "Shot": "shot",
+        "Dribble": "dribble",
+    }
+
+    shot_events = {}
+    pass_events = {}
+    dribble_events = {}
+    
+    event_mask = [event["type"]["id"] not in [5, 18, 19, 26, 27, 34, 35, 36] for event in events_json]
```

Comment:
It would be nice for readers of the code to know what you filter out here. Maybe you can create a `UNWANTED_EVENTS` map with values what they refer to. Makes it better readable and it has some kind of constants in the file which are generally easier to change than hard coded ids.

```python
UNWANTED_EVENTS = {
   35: "formations",
   ...
}
event_mask = [event["type"]["id"] not in UNWANTED_EVENTS.keys()] for event in events_json]
```

Reply:
Also, based in the `sum(event_mask)` we can calculate the number of events we have. So we can initialize the `event_data` dictionary with the corresponding not a number  values (`"datetime": [pd.NaT]*sum(event_mask)`) and in the loop use the `id` enumerator to add the event to the right spot (`event_data["datetime"][id] = pd.to_datetime(...)`). Again, to make it a bit more pythonic and not use append if we not really need it. It will also make the code a bit more clean since else blocks like `event_data["start_x"].append(np.nan)` can be removed.